### PR TITLE
[IPU] Fix Custom Poptorch options to IPUPlugin

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

When passing custom options to the IPU Plugin, I noticed that some of the options were incorrectly set leading to failures + options are not correctly set in poptorch.

This PR fixes this case where a custom IPUPlugin + options are passed in.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/accelerators/test_ipu.py [hunk 1]
```diff
@@ -460,6 +460,45 @@ def test_manual_poptorch_opts_train_grad_accum(tmpdir):
...
+    inference_opts.Training.gradientAccumulation(1)
+
+    training_opts = poptorch.Options()
+    training_opts.deviceIterations(8)
+    inference_opts.replicationFactor(2)
+    training_opts.Training.gradientAccumulation(2)
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        ipus=2,
```
Comment (davors72) [inline tests/accelerators/test_ipu.py:483]: I wonder if we should change this behaviour (of setting ipu=). It's still a bit odd because our manual pipelining/option passing leads to so many ways to run with more/less IPUs than are set here. Do you think maybe we should make this optional and if `IPUPlugin` is set we take the parameters from there?

Comment (SeanNaren) [reply tests/accelerators/test_ipu.py:483]: I agree, this is my current thinking:

`ipus=x`  is a nice way for users already familiar with lightning to get started with IPUs, but quickly this can be outgrown to swap to the `IPUPlugin`.

In addition, the IPUPlugin should override the Trainer. I.e setting poptorch.Options for gradient accumulation or the replication factor should mean that you do not need to set `Trainer(ipus=x, accumulate_grad_batches=x)`. These will be infered from the settings.

Comment (SeanNaren) [reply tests/accelerators/test_ipu.py:483]: We'll set this up in a separate PR if agreed on, as this still moves us towards these objectives.

Comment (davors72) [reply tests/accelerators/test_ipu.py:483]: Yeah I like that

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/ipu.py [hunk 2]
```diff
@@ -211,9 +211,9 @@ def process_dataloader(self, dataloader: Union[Iterable, DataLoader]) -> Union[I
             dataloader = apply_to_collection(dataloader, DataLoader, self.process_dataloader)
             return dataloader
         if not isinstance(dataloader, poptorch.DataLoader):
-            dataloader = self._convert_to_poptorch_loader(
-                dataloader=dataloader, opts=self._create_opts(training=self.lightning_module.training)
-            )
+            is_training = self.lightning_module.trainer.state.stage is RunningStage.TRAINING
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/ipu.py:214]: Shortcut

```suggestion
            is_training = self.lightning_module.trainer.training
```

**Review by ethanwharris — APPROVED** [review]

File: tests/accelerators/test_ipu.py (hunk 1 above)
Comment (SeanNaren) [reply tests/accelerators/test_ipu.py:483]: I'll handle this in a separate PR as I feel the connector will be the bottleneck, which is being addressed slowly.

**Review by carmocca — APPROVED** [review]

**Comment by SeanNaren** [comment]
This has been moved to #8241 as the caching in the CI was breaking the test.
